ims/database/db_connection.py

```diff
@@ -1,14 +1,22 @@
 from sqlalchemy import create_engine
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import sessionmaker
-from sqlalchemy.pool import NullPool
 from sqlalchemy.pool import SingletonThreadPool
 
 import ims.common.config as config
+import ims.common.constants as constants
 
 _cfg = config.get()
 
 
+# ISSUE
+# This Class is not creating the engine once as previously expected
+# Although there is no harm in creating the engine and session maker multiple
+# times, it doesnt make sense when it comes to multi threading stuff as there
+# is supposed to be one engine and one session per request.
+
+# Will Probably need to build Engine Abstraction To Solve it properly
+
```

Comment:
We didn't agree on it. I found while testing that the DB class is being loaded each time a request is coming, so as a side effect a new engine is being created each time. To do it the right way, there should have one engine and multiple sessions or else I fear we may not have consistency.

If you remember right you didn't want any sqlalchemy code in the operations.py or higher layer. So the only solution I see is creating an DBEngine Class that will be part of the RPC Server and a New DB Connection is passed to the BMI object when it is created. If we change sqlalchemy then we just need to change the DBEngine class without need for changing any other code.